posthog/temporal/batch_exports/squash_person_overrides.py

```diff
@@ -700,10 +702,11 @@ def iter_last_n_months(self) -> collections.abc.Iterator[datetime]:
         """
         current_month = datetime.now()
 
-        for _ in range(self.last_n_months):
+        for n_month in range(0, self.offset + self.last_n_months):
             current_month = current_month.replace(day=1)
 
-            yield current_month
+            if n_month >= self.offset:
```

Comment:
We could have in theory just started from `offset` months ago instead of skipping all before offset, but the trivial algorithm for subtracting months uses this same loop, so we would be repeating code. There likely is a non-trivial algorithm that does some form of modulo 12 computation for month subtraction, but I think it's a bit pointless to introduce more complexity given that this loop will not run more than a few dozen times (worst case, we squash things since the first possible partition `197001`, which is 54 years, or like 640 months, a fraction of a second to loop in a workflow that takes hours to run).